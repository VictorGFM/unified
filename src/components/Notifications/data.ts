import { AlfredoPhotoIcon, ContactPhotoIcon } from '../../assets'
import { Notification } from '../NotificationCard/NotificationCard'

export const notifications: Notification[] = [
  {
    id: '1',
    contactName: 'Alfredo',
    contactPhoto: AlfredoPhotoIcon,
    description: 'liked your tweet.',
    time: '1h',
    socialType: 'Twitter',
  },
  {
    id: '2',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    description: 'liked your post.',
    time: '1h',
    socialType: 'Instagram',
  },
  {
    id: '3',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    description: 'shared your post.',
    time: '1h',
    socialType: 'Facebook',
  },
  {
    id: '4',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    description: 'liked your post.',
    time: '1h',
    socialType: 'Instagram',
  },
  {
    id: '5',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    description: 'liked your post.',
    time: '2h',
    socialType: 'Twitter',
  },
]
