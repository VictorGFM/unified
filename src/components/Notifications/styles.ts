import styled from 'styled-components'
import { COLORS } from '../../styles/global'
import SocialMediaButton from '../SocialMediaButton'
import { Icon, SelectedBar, SocialUnreadMessagesNumber } from '../SocialMediaButton/styles'

export const NotificationsDiv = styled.div`
  display: flex;
  flex-direction: column;
  position: absolute;
  top: 70px;
  right: 72px;

  width: 479px;
  height: 332px;
  border: 1px solid ${COLORS.GREY.THREE};
  background-color: ${COLORS.WHITE.ONE};

  border-radius: 7px;
  filter: drop-shadow(0px 4px 5px rgba(0, 0, 0, 0.25));
  box-sizing: border-box;
  z-index: 2;
`

export const NotificationsHeader = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;

  height: 42px;
  border-bottom: 1px solid ${COLORS.GREY.THREE};
`

export const NotificationsTitle = styled.span`
  width: 116px;
  margin-left: 16px;

  font-size: 20px;
  font-weight: bold;
  line-height: 23px;
`

export const SocialButtonsDiv = styled.div`
  display: flex;
  margin-left: 23px;
`

export const NotificationsSocialButton = styled(SocialMediaButton)`
  width: 42px;
  height: 42px;
  border: none;
  ${Icon} {
    width: 27px;
    height: 27px;
    margin-bottom: 4px;
  }
  ${SelectedBar} {
    left: 0;
    bottom: 0px;
    width: 43px;
    height: 5.6px;
  }
  ${SocialUnreadMessagesNumber} {
    right: 2px;
    bottom: 3px;

    width: 20px;
    height: 15px;

    font-size: 10px;
    line-height: 12px;
  }
`

export const NotificationsList = styled.div`
  overflow: auto;
  overflow-x: hidden;

  :last-child {
    border-bottom-left-radius: 7px;
    border-bottom-right-radius: 7px;
  }
`
