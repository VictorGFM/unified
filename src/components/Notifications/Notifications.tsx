import { useEffect, useRef, useState } from 'react'
import { AppIcon, FacebookIcon, InstagramIcon, TwitterIcon } from '../../assets'
import NotificationCard from '../NotificationCard'
import { SocialType } from '../SocialMediaBar/SocialMediaBar'
import {
  NotificationsDiv,
  NotificationsHeader,
  NotificationsList,
  NotificationsSocialButton,
  NotificationsTitle,
  SocialButtonsDiv,
} from './styles'
import { notifications } from './data'
import { useNavigate } from 'react-router-dom'

interface Notifications {
  handleCloseNotifications: () => void
}

const Notifications = ({ handleCloseNotifications }: Notifications) => {
  const [selectedSocial, setSelectedSocial] = useState('Unified')
  const ref = useRef(null)
  const navigate = useNavigate()

  useEffect(() => {
    const handleClickOutside = event => {
      if (ref.current && !ref.current.contains(event.target)) {
        handleCloseNotifications()
      }
    }
    document.addEventListener('mousedown', handleClickOutside)
    return () => {
      document.removeEventListener('mousedown', handleClickOutside)
    }
  }, [ref])

  const handleClickButton = (socialType: SocialType) => {
    setSelectedSocial(socialType)
  }

  const handleClickNotification = (socialType: SocialType) => {
    navigate(`/social/${socialType.toLowerCase()}`)
  }

  return (
    <NotificationsDiv ref={ref}>
      <NotificationsHeader>
        <NotificationsTitle>Notifications</NotificationsTitle>
        <SocialButtonsDiv>
          <NotificationsSocialButton
            icon={AppIcon}
            isSelected={selectedSocial === 'Unified'}
            onClick={() => handleClickButton('Unified')}
            unreadMessages={64}
          />
          <NotificationsSocialButton
            icon={InstagramIcon}
            isSelected={selectedSocial === 'Instagram'}
            onClick={() => handleClickButton('Instagram')}
            unreadMessages={2}
          />
          <NotificationsSocialButton
            icon={TwitterIcon}
            isSelected={selectedSocial === 'Twitter'}
            onClick={() => handleClickButton('Twitter')}
            unreadMessages={31}
          />
          <NotificationsSocialButton
            icon={FacebookIcon}
            isSelected={selectedSocial === 'Facebook'}
            onClick={() => handleClickButton('Facebook')}
            unreadMessages={31}
          />
        </SocialButtonsDiv>
      </NotificationsHeader>
      <NotificationsList>
        {notifications &&
          notifications.map(notification => (
            <NotificationCard
              key={notification.id}
              notification={notification}
              onClick={() => handleClickNotification(notification.socialType)}
            />
          ))}
      </NotificationsList>
    </NotificationsDiv>
  )
}

export default Notifications
