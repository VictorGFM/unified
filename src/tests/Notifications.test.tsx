import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'
import { describe, expect, it } from 'vitest'
import { render } from '@testing-library/react'
import Notifications from '../components/Notifications'

describe('<Notifications/>', () => {
  it('should show match snapshot', () => {
    const component = render(
      <Router>
        <Routes>
          <Route path="*" element={<Notifications />} />
        </Routes>
      </Router>
    )
    expect(component).toMatchSnapshot()
  })
})
