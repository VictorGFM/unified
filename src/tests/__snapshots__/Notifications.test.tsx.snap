// Vitest Snapshot v1

exports[`<Notifications/> > should show match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-crXcEl keWfMn"
      >
        <div
          class="sc-evZas QXptK"
        >
          <span
            class="sc-breuTD hpSfOq"
          >
            Notifications
          </span>
          <div
            class="sc-ksZaOG kVCpTI"
          >
            <div
              class="sc-papXJ EhaSZ sc-hAZoDl kZYwjL"
            >
              <img
                class="sc-jqUVSM gubRuo"
                src="/src/assets/icons/AppIcon.svg"
              />
              <div
                class="sc-kDDrLX HKNkb"
              />
            </div>
            <div
              class="sc-papXJ fjhlhe sc-hAZoDl kZYwjL"
            >
              <img
                class="sc-jqUVSM gubRuo"
                src="/src/assets/icons/InstagramIcon.svg"
              />
              <div
                class="sc-ftvSup gajWbD sc-iqcoie bwqaUE"
              >
                2
              </div>
            </div>
            <div
              class="sc-papXJ fjhlhe sc-hAZoDl kZYwjL"
            >
              <img
                class="sc-jqUVSM gubRuo"
                src="/src/assets/icons/TwitterIcon.svg"
              />
              <div
                class="sc-ftvSup gajWbD sc-iqcoie bwqaUE"
              >
                +30
              </div>
            </div>
            <div
              class="sc-papXJ fjhlhe sc-hAZoDl kZYwjL"
            >
              <img
                class="sc-jqUVSM gubRuo"
                src="/src/assets/icons/FacebookIcon.svg"
              />
              <div
                class="sc-ftvSup gajWbD sc-iqcoie bwqaUE"
              >
                +30
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-fnykZs gvyPkA"
        >
          <div
            class="sc-dkzDqf hXIkkB"
            data-cy="notificaiton-card-0"
          >
            <img
              class="sc-hKMtZM eaHsLj"
              src="/src/assets/icons/AlfredoPhotoIcon.svg"
            />
            <span
              class="sc-eCYdqJ jMxTVP"
            >
              Alfredo
            </span>
            <span
              class="sc-jSMfEi jCXEyz"
            >
              liked your tweet.
            </span>
            <span
              class="sc-gKXOVf fVYPhJ"
            >
              1h
            </span>
            <div
              class="sc-bczRLJ gpaXLH"
            >
              <img
                class="sc-gsnTZi gNNCuJ"
                src="/src/assets/icons/TwitterIcon.svg"
              />
            </div>
          </div>
          <div
            class="sc-dkzDqf hXIkkB"
            data-cy="notificaiton-card-1"
          >
            <img
              class="sc-hKMtZM eaHsLj"
              src="/src/assets/icons/ContactPhotoIcon.svg"
            />
            <span
              class="sc-eCYdqJ jMxTVP"
            >
              Contact Name
            </span>
            <span
              class="sc-jSMfEi jCXEyz"
            >
              liked your post.
            </span>
            <span
              class="sc-gKXOVf fVYPhJ"
            >
              1h
            </span>
            <div
              class="sc-bczRLJ gpaXLH"
            >
              <img
                class="sc-gsnTZi gNNCuJ"
                src="/src/assets/icons/InstagramIcon.svg"
              />
            </div>
          </div>
          <div
            class="sc-dkzDqf hXIkkB"
            data-cy="notificaiton-card-2"
          >
            <img
              class="sc-hKMtZM eaHsLj"
              src="/src/assets/icons/ContactPhotoIcon.svg"
            />
            <span
              class="sc-eCYdqJ jMxTVP"
            >
              Contact Name
            </span>
            <span
              class="sc-jSMfEi jCXEyz"
            >
              shared your post.
            </span>
            <span
              class="sc-gKXOVf fVYPhJ"
            >
              1h
            </span>
            <div
              class="sc-bczRLJ gpaXLH"
            >
              <img
                class="sc-gsnTZi gNNCuJ"
                src="/src/assets/icons/FacebookIcon.svg"
              />
            </div>
          </div>
          <div
            class="sc-dkzDqf hXIkkB"
            data-cy="notificaiton-card-3"
          >
            <img
              class="sc-hKMtZM eaHsLj"
              src="/src/assets/icons/ContactPhotoIcon.svg"
            />
            <span
              class="sc-eCYdqJ jMxTVP"
            >
              Contact Name
            </span>
            <span
              class="sc-jSMfEi jCXEyz"
            >
              liked your post.
            </span>
            <span
              class="sc-gKXOVf fVYPhJ"
            >
              1h
            </span>
            <div
              class="sc-bczRLJ gpaXLH"
            >
              <img
                class="sc-gsnTZi gNNCuJ"
                src="/src/assets/icons/InstagramIcon.svg"
              />
            </div>
          </div>
          <div
            class="sc-dkzDqf hXIkkB"
            data-cy="notificaiton-card-4"
          >
            <img
              class="sc-hKMtZM eaHsLj"
              src="/src/assets/icons/ContactPhotoIcon.svg"
            />
            <span
              class="sc-eCYdqJ jMxTVP"
            >
              Contact Name
            </span>
            <span
              class="sc-jSMfEi jCXEyz"
            >
              liked your post.
            </span>
            <span
              class="sc-gKXOVf fVYPhJ"
            >
              2h
            </span>
            <div
              class="sc-bczRLJ gpaXLH"
            >
              <img
                class="sc-gsnTZi gNNCuJ"
                src="/src/assets/icons/TwitterIcon.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-crXcEl keWfMn"
    >
      <div
        class="sc-evZas QXptK"
      >
        <span
          class="sc-breuTD hpSfOq"
        >
          Notifications
        </span>
        <div
          class="sc-ksZaOG kVCpTI"
        >
          <div
            class="sc-papXJ EhaSZ sc-hAZoDl kZYwjL"
          >
            <img
              class="sc-jqUVSM gubRuo"
              src="/src/assets/icons/AppIcon.svg"
            />
            <div
              class="sc-kDDrLX HKNkb"
            />
          </div>
          <div
            class="sc-papXJ fjhlhe sc-hAZoDl kZYwjL"
          >
            <img
              class="sc-jqUVSM gubRuo"
              src="/src/assets/icons/InstagramIcon.svg"
            />
            <div
              class="sc-ftvSup gajWbD sc-iqcoie bwqaUE"
            >
              2
            </div>
          </div>
          <div
            class="sc-papXJ fjhlhe sc-hAZoDl kZYwjL"
          >
            <img
              class="sc-jqUVSM gubRuo"
              src="/src/assets/icons/TwitterIcon.svg"
            />
            <div
              class="sc-ftvSup gajWbD sc-iqcoie bwqaUE"
            >
              +30
            </div>
          </div>
          <div
            class="sc-papXJ fjhlhe sc-hAZoDl kZYwjL"
          >
            <img
              class="sc-jqUVSM gubRuo"
              src="/src/assets/icons/FacebookIcon.svg"
            />
            <div
              class="sc-ftvSup gajWbD sc-iqcoie bwqaUE"
            >
              +30
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-fnykZs gvyPkA"
      >
        <div
          class="sc-dkzDqf hXIkkB"
          data-cy="notificaiton-card-0"
        >
          <img
            class="sc-hKMtZM eaHsLj"
            src="/src/assets/icons/AlfredoPhotoIcon.svg"
          />
          <span
            class="sc-eCYdqJ jMxTVP"
          >
            Alfredo
          </span>
          <span
            class="sc-jSMfEi jCXEyz"
          >
            liked your tweet.
          </span>
          <span
            class="sc-gKXOVf fVYPhJ"
          >
            1h
          </span>
          <div
            class="sc-bczRLJ gpaXLH"
          >
            <img
              class="sc-gsnTZi gNNCuJ"
              src="/src/assets/icons/TwitterIcon.svg"
            />
          </div>
        </div>
        <div
          class="sc-dkzDqf hXIkkB"
          data-cy="notificaiton-card-1"
        >
          <img
            class="sc-hKMtZM eaHsLj"
            src="/src/assets/icons/ContactPhotoIcon.svg"
          />
          <span
            class="sc-eCYdqJ jMxTVP"
          >
            Contact Name
          </span>
          <span
            class="sc-jSMfEi jCXEyz"
          >
            liked your post.
          </span>
          <span
            class="sc-gKXOVf fVYPhJ"
          >
            1h
          </span>
          <div
            class="sc-bczRLJ gpaXLH"
          >
            <img
              class="sc-gsnTZi gNNCuJ"
              src="/src/assets/icons/InstagramIcon.svg"
            />
          </div>
        </div>
        <div
          class="sc-dkzDqf hXIkkB"
          data-cy="notificaiton-card-2"
        >
          <img
            class="sc-hKMtZM eaHsLj"
            src="/src/assets/icons/ContactPhotoIcon.svg"
          />
          <span
            class="sc-eCYdqJ jMxTVP"
          >
            Contact Name
          </span>
          <span
            class="sc-jSMfEi jCXEyz"
          >
            shared your post.
          </span>
          <span
            class="sc-gKXOVf fVYPhJ"
          >
            1h
          </span>
          <div
            class="sc-bczRLJ gpaXLH"
          >
            <img
              class="sc-gsnTZi gNNCuJ"
              src="/src/assets/icons/FacebookIcon.svg"
            />
          </div>
        </div>
        <div
          class="sc-dkzDqf hXIkkB"
          data-cy="notificaiton-card-3"
        >
          <img
            class="sc-hKMtZM eaHsLj"
            src="/src/assets/icons/ContactPhotoIcon.svg"
          />
          <span
            class="sc-eCYdqJ jMxTVP"
          >
            Contact Name
          </span>
          <span
            class="sc-jSMfEi jCXEyz"
          >
            liked your post.
          </span>
          <span
            class="sc-gKXOVf fVYPhJ"
          >
            1h
          </span>
          <div
            class="sc-bczRLJ gpaXLH"
          >
            <img
              class="sc-gsnTZi gNNCuJ"
              src="/src/assets/icons/InstagramIcon.svg"
            />
          </div>
        </div>
        <div
          class="sc-dkzDqf hXIkkB"
          data-cy="notificaiton-card-4"
        >
          <img
            class="sc-hKMtZM eaHsLj"
            src="/src/assets/icons/ContactPhotoIcon.svg"
          />
          <span
            class="sc-eCYdqJ jMxTVP"
          >
            Contact Name
          </span>
          <span
            class="sc-jSMfEi jCXEyz"
          >
            liked your post.
          </span>
          <span
            class="sc-gKXOVf fVYPhJ"
          >
            2h
          </span>
          <div
            class="sc-bczRLJ gpaXLH"
          >
            <img
              class="sc-gsnTZi gNNCuJ"
              src="/src/assets/icons/TwitterIcon.svg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
